assets/admin/students/student-bulk-action-button/index.js

```diff
@@ -0,0 +1,79 @@
+/**
+ * WordPress dependencies
+ */
+import { Button } from '@wordpress/components';
+import { render, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import StudentModal from '../student-modal';
+
+/**
+ * Student bulk action button.
+ */
+export const StudentsBulkActionButton = () => {
+	const [ action, setAction ] = useState( 'add' );
+	const [ isModalOpen, setIsModalOpen ] = useState( false );
+	const [ studentIds, setStudentIds ] = useState( [] );
+	const closeModal = () => setIsModalOpen( false );
+	const setActionValue = ( selectedValue ) => {
+		switch ( selectedValue ) {
+			case 'enrol_restore_enrolment':
+				setAction( 'add' );
+				break;
+			case 'remove_enrolment':
+				setAction( 'remove' );
+				break;
+			case 'remove_progress':
+				setAction( 'reset-progress' );
+				break;
+			default:
+		}
+	};
+
+	const openModal = () => {
+		const hiddenSenseiBulkAction = document.getElementById(
```

Comment:
This feels like an anti-pattern. I wonder if we should convert the dropdown to React as well. WDYT @gabrielcaires?

Reply:
Good question, I think it depends.

If we want, in the future, to convert the entire page to react and we have time,  I would say that we should convert the drop-downs to `react` more due to these reasons than an `Anti-pattern`

If we are going to continue integrating react components with already existent HTML/jQuery Code, I would say that the way that we are integrating both might be different.

I think that we could move the responsibility to open modals to a single component that also exports a global function that calls be called by React/jQuery/Vanilla Js and it is responsible to open and close the modal, using the proper params.

Something like `window.openStudentModal = (students, action) => // logic to open the modal`

And instead of repeating the logic:

```
{ isModalOpen && (
				<StudentModal
					action={ action }
					onClose={ closeModal }
					students={ [ studentId ] }
				/>
			) }
	```
on every component that we want to open a modal, we can just call `window.openStudentModal` with the appropriate params.

Or using an approach like it https://www.falldowngoboone.com/blog/talk-to-your-react-components-with-custom-events/

I think adopting it can help us to keep this kind of logic:
```
https://github.com/Automattic/sensei/blob/2bf5e8d01a04d0028e9fa17b5fdcfa2d992530b0/assets/admin/students/student-bulk-action-button/index.js#L37-L51
```

Outside of our React components, so we can just change the `onClick` on the `Select Courses` button to get the selected ids (probably using jQuery) and call our global function to open the modal.

Considering we already have some PRs merged about it, I  would suggest to keep the current approach, and as soon we have more complexity, we move to another approach.

What do you think @donnapep  and @LavaGolem ?

Reply:
> If we are going to continue integrating react components with already existent HTML/jQuery Code, I would say that the way that we are integrating both might be different.

At this point, there isn't really a compelling reason for us to rewrite the Students page to use React. If we need to build brand new pages that don't exist already, then it would be good to seriously consider using React for it.

> Considering we already have some PRs merged about it, I would suggest keeping the current approach, and as soon we have more complexity, we move to another approach.

Makes sense to me! This is also something we could consider refactoring when we're on janitorial if we want.

@LavaGolem If you agree with all of the above, then there's no need to worry about eliminating instances of `document.getElementById` in the component.